From 19b4bc9d128f770dc7149399bd95f3367e2bc18c Mon Sep 17 00:00:00 2001
From: Juho Kilpikoski <juho.kilpikoski@iki.fi>
Date: Sat, 27 Mar 2010 13:36:02 +0200
Subject: [PATCH 1/3] =?UTF-8?q?Lis=C3=A4=C3=A4=20metriikoiden=20seuranta?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 milloinviimeksi/Rakefile |   32 ++++++++++++++++++++++++++++++++
 1 files changed, 32 insertions(+), 0 deletions(-)

diff --git a/milloinviimeksi/Rakefile b/milloinviimeksi/Rakefile
index 3bb0e85..3d935e6 100644
--- a/milloinviimeksi/Rakefile
+++ b/milloinviimeksi/Rakefile
@@ -8,3 +8,35 @@ require 'rake/testtask'
 require 'rake/rdoctask'
 
 require 'tasks/rails'
+
+require 'metric_fu'
+
+MetricFu::Configuration.run do |config|
+    #define which metrics you want to use
+    config.metrics  = [:churn, :saikuro, :stats, :flog, :flay, :reek, :roodi]#, :rcov]
+   config.graphs   = [:flog, :flay, :reek, :roodi]#, :rcov]
+    # config.flay     = { :dirs_to_flay => ['app', 'lib'],
+    #                     :minimum_score => 100  } 
+    # config.flog     = { :dirs_to_flog => ['app', 'lib']  }
+    # config.reek     = { :dirs_to_reek => ['app', 'lib']  }
+    # config.roodi    = { :dirs_to_roodi => ['app', 'lib'] }
+    # config.saikuro  = { :output_directory => 'scratch_directory/saikuro', 
+    #                     :input_directory => ['app', 'lib'],
+    #                     :cyclo => "",
+    #                     :filter_cyclo => "0",
+    #                     :warn_cyclo => "5",
+    #                     :error_cyclo => "7",
+    #                     :formater => "text"} #this needs to be set to "text"
+    # config.churn    = { :start_date => "1 year ago", :minimum_churn_count => 10}
+    # config.rcov     = { :environment => 'test',
+    #                     :test_files => ['test/**/*_test.rb', 
+    #                                     'spec/**/*_spec.rb'],
+    #                     :rcov_opts => ["--sort coverage", 
+    #                                    "--no-html", 
+    #                                    "--text-coverage",
+    #                                    "--no-color",
+    #                                    "--profile",
+    #                                    "--rails",
+    #                                    "--exclude /gems/,/Library/,spec"]}
+    # config.graph_engine = :bluff
+end
\ No newline at end of file
-- 
1.7.0.3

